README.MD

```diff
@@ -1,105 +1,222 @@
+
+Redeeming:
 ```js
-let auxiliaryConfig = {
-  gasPrice: '0',
-  tokenOrganization: {
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-    owner: "0x...", /* Address of owner key of Organization */
-    admin: "0x...", /* Address of admin key of Organization */
-    workers: ["0x..."] /* Address for Organized Workers */
-  },
-  ostPrime: {
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-    chainOwner: config.chainOwner
-  },
-  anchorOrganization: {
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-    owner: "0x...", /* Address of owner key of Organization */
-    admin: "0x...", /* Address of admin key of Organization */
-    workers: ["0x..."] /* Address for Organized Workers */
-  },
-  anchor: {
-    remoteChainId: '12345', /* Origin chain-id */
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-    organizationOwner: "0x..." /* Address of owner/admin key of Organization */
-  },
-  libs: {
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-  },
-  cogateway: {
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-    "bounty": '1000000000000000000'
-  }
+const redeemer = '0x0000000000000000000000000redeemeraddress';
+const redeemAmount = '10000';
+const beneficiary = '0x000000000000000beneficiaryaccountaddress';
+const gasPrice = '1';
+const gasLimit = '10000000';
+const hashLock = '00000000000000000000000000000000000000000000000000000000hashlock';
+const txOptionRedeem = {
+  from: redeemer,
+  value: '100', // This must be equal to the bounty amount!
 };
+
+facilitator
+  .redeem(
+    redeemer,
+    redeemAmount,
+    beneficiary,
+    gasPrice,
+    gasLimit,
+    hashLock,
+    txOptionRedeem,
+  )
+  .then(console.log)
+  .catch(console.log);
 ```
-### Executing setup. This may some time.
+
+Progressing the redemption after the state root has been transferred:
 ```js
-helper.setup(simpleToken, originConfig, auxiliaryConfig)
-.then(function (output) {
-  console.log("output", output);
-})
+const unlockSecret = '0x0000000000000000000000000000000000000000000000000000unlocksecret';
+const txOptionOriginRedeem = { from: '0x000000000facilitatoraddressonoriginchain' };
+const txOptionAuxiliaryRedeem = { from: '0x000000facilitatoraddressonauxiliarychain' };
+
+facilitator
+  .progressRedeem(
+    redeemer,
+    nonce,
+    beneficiary,
+    redeemAmount,
+    gasPrice,
+    gasLimit,
+    hashLock,
+    unlockSecret,
+    txOptionOriginRedeem,
+    txOptionAuxiliaryRedeem,
+  )
+  .then(console.log)
+  .catch(console.log);
 ```
 
+#### Deploying Gateways
+
+Using `mosaic.js` to deploy gateways is still a work in progress and must be done manually for now:
+```js
+// Expecting Mosaic impart and an instance of `mosaic`:
+const { EIP20Gateway } = Mosaic.Contracts;
+
+EIP20Gateway
+  .setup(
+    mosaic.origin.web3,
+    mosaic.auxiliary.web3,
+    gatewayConfig,
```

Comment:
The contents of `gatewayConfig` and `coGatewayConfig` are probably the things that need the most explanation about the setup, and without that I'm not sure if the docs here provide much value.

Maybe add we should at least add a link to the typedef of the configs in the code until we have rendered & hosted docs we can link to?

Reply:
Yeah it is difficult. I had objects and short explanations, but it results in even more questions. 
Where do I get the addresses from? How do I deploy them if I need to?

Maybe we should remove the whole section from the readme. It is not ready, yet.